explore/src/main/scala/explore/config/SpectroscopyModesTable.scala

```diff
@@ -218,35 +211,77 @@ object SpectroscopyModesTable {
   ): Boolean =
     rowToConf(row).exists(_ === conf)
 
-  protected val tableComponent =
-    ScalaFnComponent[ModesTableProps] { props =>
-      def toggleRow(row: SpectroscopyModeRow): Option[ScienceConfigurationData] =
-        rowToConf(row).filterNot(conf => props.scienceConfiguration.get.contains_(conf))
-
-      val tableInstance = ModesTableMaker.use(props.options)
-      <.div(
-        ExploreStyles.ModesTable,
-        ModesTableComponent(
-          table =
-            Table(celled = true, selectable = true, striped = true, compact = TableCompact.Very)(),
-          header = true,
-          headerCell = (c: ModesTableMaker.ColumnType) =>
-            TableHeaderCell(clazz = ExploreStyles.Sticky |+| ExploreStyles.ModesHeader)(
-              ^.textTransform.capitalize.when(c.id.toString =!= ResolutionColumnId.value),
-              ^.textTransform.none.when(c.id.toString === ResolutionColumnId.value)
-            ),
-          row = (rowData: Row[SpectroscopyModeRow]) =>
-            TableRow(
-              disabled = !enabledRow(rowData.original),
-              clazz = ExploreStyles.ModeSelected.when_(
-                props.scienceConfiguration.get.exists(conf => equalsConf(rowData.original, conf))
+  // protected def ensureContains(matrix: List[SpectroscopyModeRow], conf: Option[ScienceConfigurationData]): List[SpectroscopyModeRow] =
```

Comment:
Commented code

Reply:
Yup, we will probably use it to show the selected config when it's not in the list.